cyral/resource_cyral_repository.go

```diff
@@ -429,15 +440,31 @@ func resourceRepository() *schema.Resource {
 							Optional:    true,
 						},
 						RepoMongoDBServerTypeKey: {
-							Description:  "Type of the MongoDB server. Allowed values: " + supportedTypesMarkdown(mongoServerTypes()),
+							Description: "Type of the MongoDB server. Allowed values: " + supportedTypesMarkdown(mongoServerTypes()) +
+								"\n\n  The following conditions apply:\n" +
+								"  - If `" + Sharded + "` and `" + RepoMongoDBSRVRecordName + "` *not* provided, then all `" +
+								RepoNodesKey + "` blocks must be static (see [`" + RepoNodeDynamicKey + "`](#" + RepoNodeDynamicKey + ")).\n" +
+								"  - If `" + Sharded + "` and `" + RepoMongoDBSRVRecordName + "` provided, then all `" +
+								RepoNodesKey + "` blocks must be dynamic (see [`" + RepoNodeDynamicKey + "`](#" + RepoNodeDynamicKey + ")).\n" +
+								"  - If `" + Standalone + "`, then only one `" + RepoNodesKey +
+								"` block can be declared and it must be static (see [`" + RepoNodeDynamicKey + "`](#" + RepoNodeDynamicKey + ")). The `" +
+								RepoMongoDBSRVRecordName + "` is not supported in this configuration.\n" +
+								"  - If `" + ReplicaSet + "` and `" + RepoMongoDBSRVRecordName + "` *not* provided, then `" +
+								RepoNodesKey + "` blocks may mix dynamic and static nodes (see [`" + RepoNodeDynamicKey + "`](#" + RepoNodeDynamicKey + ")).\n" +
+								"  - If `" + ReplicaSet + "` and `" + RepoMongoDBSRVRecordName + "` provided, then `" +
+								RepoNodesKey + "` blocks must be dynamic (see [`" + RepoNodeDynamicKey + "`](#" + RepoNodeDynamicKey + ")).\n",
 							Type:         schema.TypeString,
 							Required:     true,
 							ValidateFunc: validation.StringInSlice(mongoServerTypes(), false),
 						},
 						RepoMongoDBSRVRecordName: {
-							Description: "Name of a DNS SRV record which contains cluster topology details",
-							Type:        schema.TypeString,
-							Optional:    true,
+							Description: "Name of a DNS SRV record which contains cluster topology details. " +
```

Comment:
This should default to an empty string. The backend will interpret that as "no SRV record supplied" and thus it will go through the "hard-coded" path.